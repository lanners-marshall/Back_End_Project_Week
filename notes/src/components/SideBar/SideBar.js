import React from 'react';
import { Link } from "react-router-dom";
import { Aside, DivClick, Log, NewsH2, NewsDiv, NewsOrg } from './css';
import styled from 'styled-components';

class SideBar extends React.Component {
	constructor(props){
		super(props);
		this.state = {};
	}

	logout = event => {
		localStorage.removeItem('error');
    localStorage.removeItem('jwt');
    this.props.history.push('/signin')
  }

	render() {
		return (	
			<Aside>
				<h1>Lambda<br /> Notes</h1>
				<Link to="/notes"><DivClick><p>View Your Notes</p></DivClick></Link>
				<Link to="/notes/create"><DivClick><p>+ Create New Note</p></DivClick></Link>
				<DivClick onClick={this.logout}><p>Log Out</p></DivClick>
				<NewsH2>Current News</NewsH2>
				{this.props.news.map((n, i )=> (
					<NewsDiv key={i}>
						<a href={n.url}><p>{n.title}</p></a>
					</NewsDiv>
				))}
				<a href="http://newsapi.org"><NewsOrg>- newsapi.org</NewsOrg></a>
			</Aside>
		)
	}
}

export default SideBar;

