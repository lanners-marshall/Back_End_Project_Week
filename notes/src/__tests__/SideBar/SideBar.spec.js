import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter, Route } from 'react-router-dom';

import SideBar from '../../components/SideBar/SideBar';
import renderer from 'react-test-renderer';

describe('<SideBar />', () => {
  it('should match snapshot', () => {
  	let news =[{title: 'news-title', url: 'https://www.test-example.com'},{title: 'news-title', url: 'https://www.test-example.com'}]
	  const wrapper = mount(
	  	<MemoryRouter initialEntries={['/notes']}>
	      <SideBar news={news} />
	    </MemoryRouter>
  	)
  	const tree = renderer.create(wrapper).toJSON()
  	expect(tree).toMatchSnapshot()
	})
});