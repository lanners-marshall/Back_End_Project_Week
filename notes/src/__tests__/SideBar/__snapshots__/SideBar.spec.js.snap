// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should match snapshot 1`] = `
<div
  className="sc-bdVaJa eUgSDP"
>
  <h1>
    Lambda
    <br />
     Notes
  </h1>
  <a
    href="/notes"
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH dZtenF"
    >
      <p>
        View Your Notes
      </p>
    </div>
  </a>
  <a
    href="/notes/create"
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH dZtenF"
    >
      <p>
        + Create New Note
      </p>
    </div>
  </a>
  <div
    className="sc-bwzfXH dZtenF"
    onClick={[Function]}
  >
    <p>
      Log Out
    </p>
  </div>
  <h2
    className="sc-bxivhb fYvkCU"
  >
    Current News
  </h2>
  <div
    className="sc-ifAKCX dBEyXr"
  >
    <a
      href="https://www.test-example.com"
    >
      <p>
        news-title
      </p>
    </a>
  </div>
  <div
    className="sc-ifAKCX dBEyXr"
  >
    <a
      href="https://www.test-example.com"
    >
      <p>
        news-title
      </p>
    </a>
  </div>
  <a
    href="http://newsapi.org"
  >
    <p
      className="sc-EHOje ksLjHH"
    >
      - newsapi.org
    </p>
  </a>
</div>
`;
